app/controllers/users_controller.rb

```diff
@@ -1,6 +1,8 @@
 class UsersController < ApplicationController
-  before_action :check_is_logged_in, :check_is_admin_permission
-  before_action :get_user, only: [:edit, :update, :destroy, :show]
+  before_action :check_is_logged_in, :check_is_admin_permission,
+    only: [:index, :show, :new, :create, :edit, :update, :destroy]
+  before_action :check_is_logged_in, only: [:edit_profile]
```

Comment:
`check_is_logged_in` applies for all method so it doesn't need `:only` filter
`check_is_admin_permission` you can use blacklist filter, `except: [:edit_profile]`  instead of `only: [:index, :show, :new, :create, :edit, :update, :destroy]`